active/0000-routeable-components.md

```diff
@@ -0,0 +1,488 @@
+method.
+
+We will add a complementary `update` action that fires after your
+`attributes` hook only when the route has not been initially entered.
+
+
+### Loading and error states
+
+`Routes` will still fire the `loading` and `error` actions as they do
+now.
+
+The default handlers for those actions currently look for
+correspondingly-named templates. We would change them to look for
+correspondingly-named components instead. For example, if before you
+had a template named `'people/loading'`, you would instead have a
+component named `'people/loading'`.
+
+This is actually a fairly small change, since adding a template is
+sufficient to define a component. But it unifies the API, and gives
+you the full capability of components for implementing these
+substates.
+
+### Sending actions from component up to route
+
+Components are deliberately isolated and don't send arbitrary actions
+upward unless you explicitly tell them to. Routable components
+preserve this same behavior.
+
+Therefore, if you want to send an action from a component up to the
+Route that rendered it, you should pass an action handler function as
+one of your component's attributes.
+
+### Fragments: "Tagless Components"
+
+Today, when you render a template into an outlet, neither the template
+nor the outlet introduces extra DOM elements. We want to be able to do
+the same thing with routable components, which means we need to
+officially support "tagless components".
+
+But instead of having two flavors of components (tagged and tagless)
+with subtly different behavior, we will introduce a new concept:
+`Ember.Fragment`. A Fragment is a "tagless component". It still
+receives attributes and can maintain internal state, and it renders a
+template, but it does not have a top-level element and it may generate
+zero, one, or more DOM elements.
+
+`Fragments` have no `element` property. They *do* have a `$` property,
+but it represents a DOM range, not a single element.
+
+Everywhere else in this document where I say "component", I really
+mean `Component` *or* `Fragment`.
+
+To distinguish `Component` templates from `Fragment` templates, we
+will introduce the `<element>` and `<fragment>` keywords, which are
+described in a [separate RFC document](element-and-fragment).
+
+## Query Parameters
+
+Today, query parameters live on controllers, which means we need to
+find them a new home. Neither Routes nor Components are intended to be
+stateful, so we will not bind the values of queryParams directly to
+either of those.
+
+queryParams will be *declared and configured* on Routes, with the same
+API we currently have on Controllers except for two exceptions. The
+first exception is default values -- a query parameter like this:
+
+````js
+Ember.Controller.extend({
+  queryParams: {
+    foo: { as: 'bar' }
+  }
+  bar: 0
+});
+````
+
+would look like this on a Route:
+
+````js
+  queryParams: {
+    foo: { as: 'bar', default: 0 }
```

Comment:
we should describe the behaviour around unsetting and re-defaulting if it the value becomes `undefined`